src/ContainerAppHelper.ts

```diff
@@ -0,0 +1,440 @@
+                const lines = data.split('\n');
+                return lines[0];
+            }).catch((err) => {
+                core.setFailed(err.message);
+                throw err;
+            });
+
+            // Delete the temp file
+            fs.unlink(oryxRuntimeTxtPath, (err) => {
+                if (err) {
+                    core.warning(`Unable to delete the temporary file "${oryxRuntimeTxtPath}". Error: ${err.message}`);
+                }
+            });
+
+            return runtimeStack;
+        } catch (err) {
+            core.setFailed(err.message);
+            throw err;
+        }
+    }
+
+    /**
+     * Sets the default builder on the machine to the Oryx++ Builder to prevent an exception from being thrown due
+     * to no default builder set.
+     */
+    public async setDefaultBuilder() {
+        core.info('Setting the Oryx++ Builder as the default builder via the pack CLI');
+        try {
+            await new Utility().executeAndthrowIfError(`${PACK_CMD}`, ['config', 'default-builder', `${ORYX_BUILDER_IMAGE}`]);
+        }
+        catch (err) {
+            core.setFailed(err.message);
+            throw err;
+        }
+    }
+
+    /**
+     * Installs the pack CLI that will be used to build a runnable application image.
+     * For more information about the pack CLI can be found here: https://buildpacks.io/docs/tools/pack/
+     */
+    public async installPackCliAsync() {
+        core.debug('Attempting to install the pack CLI');
+        try {
+            let command: string = '';
+            let commandLine = '';
+            let args: string[] = [];
+            if (IS_WINDOWS_AGENT) {
+                const packZipDownloadUri: string = 'https://github.com/buildpacks/pack/releases/download/v0.27.0/pack-v0.27.0-windows.zip';
+                const packZipDownloadFilePath: string = path.join(PACK_CMD, 'pack-windows.zip');
+                args = [`New-Item`, `-ItemType`, `Directory`, `-Path`, `${PACK_CMD}`, `-Force | Out-Null;`, `Invoke-WebRequest`, `-Uri`, `${packZipDownloadUri}`, `-OutFile`, `${packZipDownloadFilePath};`, `Expand-Archive`, `-LiteralPath`, `${packZipDownloadFilePath}`, `-DestinationPath`, `${PACK_CMD};`, `Remove-Item`, `-Path`, `${packZipDownloadFilePath}`,
+                       `Expand-Archive`, `-LiteralPath`, `${packZipDownloadFilePath}`, `-DestinationPath`, `${PACK_CMD};`, `Remove-Item`, `-Path`, `${packZipDownloadFilePath}`];
+                commandLine = 'pwsh';
+            } else {
+                const tgzSuffix = os.platform() == 'darwin' ? 'macos' : 'linux';
+                command = `(curl -sSL \"https://github.com/buildpacks/pack/releases/download/v0.27.0/pack-v0.27.0-${tgzSuffix}.tgz\" | ` +
+                    'tar -C /usr/local/bin/ --no-same-owner -xzv pack)';
+                args = ['-c', command];
+                commandLine = 'bash';
+            }
+            await new Utility().executeAndthrowIfError(commandLine, args);
```

Comment:
Even though the documentation mentions `@actions/exec` is platform agnostic, I'm not sure if the `exec` method can accept the command to be executed as the only arg and determine the `os` dynamically. The command fails by just passing the `command to be executed` as the first arg.
 From examples in [this doc](https://github.com/actions/toolkit/blob/main/packages/exec/__tests__/exec.test.ts#L69), looks like the first argument must be the command line it self like 'docker' or 'bash' and the second argument of `exec` are the args to the command line. This [example](https://github.com/actions/toolkit/blob/main/packages/exec/__tests__/exec.test.ts#L35), passes arguments to the appropriate command line after checking what the `os` is.